### ISSUE TITLE
TabBarView should give priority to vertical scrolling over horizontal

### ISSUE THREAD

**Issue by najeira** [flutter/flutter#11132]
I faced a problem when using TabBarView that includes multiple ScrollViews.

I try to scroll vertically, but sometimes fail. It is because the thumb moves diagonally when operating with one hand.

I want to prioritize vertical (ie inner) scroll than horizontal. Because in mobile apps, vertical scrolling is more frequently operation than horizontal.

TabBarView uses HorizontalDragGestureRecognizer internally. It judges whether the delta of horizontal movement is sufficient.

https://github.com/flutter/flutter/blob/f64bfba860d143f3f3b9d8970a15c088b0d9599d/packages/flutter/lib/src/gestures/monodrag.dart#L259

The delta is same value in both VerticalDragGestureRecognizer and HorizontalDragGestureRecognizer. I think that changing that value will work for TabBarView/PageView.

In addition, I found `kPagingTouchSlop`.

https://github.com/flutter/flutter/blob/438f4df00c64c304d037b7cf3d6678199816ab18/packages/flutter/lib/src/gestures/constants.dart#L62

```dart
/// The distance a touch has to travel for us to be confident that the gesture
/// is a paging gesture. (Currently not used, because paging uses a regular drag
/// gesture, which uses kTouchSlop.)
// TODO(ianh): Create variants of HorizontalDragGestureRecognizer et al for
// paging, which use this constant.
const double kPagingTouchSlop = kTouchSlop * 2.0;  // Logical pixels
```

## Flutter Doctor

```
[✓] Flutter (on Mac OS X 10.12.5 16F73, locale ja-JP, channel master)
    • Flutter at /Applications/flutter
    • Framework revision 4891506931 (2 days ago), 2017-07-07 16:00:45 -0700
    • Engine revision 8ad1576f27
    • Tools Dart version 1.25.0-dev.4.0
```

**Comment by eseidelGoogle** [comment]
I'm not sure what to do with this bug, but @Hixie or @HansMuller would know.

**Comment by Hixie** [comment]
Yeah the fix here is to introduce a variant of HorizontalDragGestureRecognizer that's specifically for paging (maybe just making it an argument to the constructor, or a second constructor that configures the recognizer slightly differently), and then have the page views use that one instead. Should be relatively straight-forward.

**Comment by PaulDelestrac** [comment]
Would it be possible to use both horizontal and vertical scrolling at the same time? 
I don't know if what I'm saying is making any sense so tell me if you understand 😅

**Comment by cwebbtw** [comment]
Is there any workaround for this?

**Comment by mattprinold** [comment]
What is the status of this as the issue still exists and it's been a long time since this was raised?
